addons/network/functions/fnc_initRouter.sqf

```diff
@@ -37,53 +37,54 @@ private _childs =
 	_actions
 };
 
-private _connect = ["AE3_Network_ConnectAction", localize "STR_AE3_Network_Interaction_ConnectToRouter", "",
-			{},
-			{
-				params ["_target", "_player", "_params"]; 
-				_params params ["_device"]; 
-				(alive _target) and (isNull (_device getVariable ["AE3_network_parent", objNull]))
-			},
-			_childs,
-			[_entity]
-			] call ace_interact_menu_fnc_createAction;
-
-private _disconnect = ["AE3_Network_DisconnectAction", localize "STR_AE3_Network_Interaction_DisconnectFromRouter", "",
+if (!isDedicated && !_internal) then
+{
+	private _connect = ["AE3_Network_ConnectAction", localize "STR_AE3_Network_Interaction_ConnectToRouter", "",
+				{},
 				{
 					params ["_target", "_player", "_params"]; 
 					_params params ["_device"]; 
-					[_device] call AE3_network_fnc_disconnect;
-				},
-				{
-					params ["_target", "_player", "_params"]; 
-					_params params ["_device"];
-					(alive _target) and (!isNull (_device getVariable ["AE3_network_parent", objNull]))
+					(alive _target) and (isNull (_device getVariable ["AE3_network_parent", objNull]))
 				},
-				{},
+				_childs,
 				[_entity]
 				] call ace_interact_menu_fnc_createAction;
 
-
-if (!isDedicated && !_internal) then
-{
-	[_entity, 0, ["ACE_MainActions", "AE3_DeviceAction"], _connect] call ace_interact_menu_fnc_addActionToObject;
-	[_entity, 0, ["ACE_MainActions", "AE3_DeviceAction"], _disconnect] call ace_interact_menu_fnc_addActionToObject;
+	private _disconnect = ["AE3_Network_DisconnectAction", localize "STR_AE3_Network_Interaction_DisconnectFromRouter", "",
+					{
+						params ["_target", "_player", "_params"]; 
+						_params params ["_device"]; 
+						[_device] call AE3_network_fnc_disconnect;
+					},
+					{
+						params ["_target", "_player", "_params"]; 
+						_params params ["_device"];
+						(alive _target) and (!isNull (_device getVariable ["AE3_network_parent", objNull]))
+					},
+					{},
+					[_entity]
+					] call ace_interact_menu_fnc_createAction;
+
+	private _parentActionPath = _entity getVariable ["AE3_parentActionPath", ""];
```

Comment:
Why don't u use addParentAction instead?

Reply:
I don't have access to the `_name` variable. To achieve that I would need to pass additional parameters to this function. Because of the way the code flows, this variable is already set and can easily be used.